src/lib/state/api.ts

```diff
@@ -89,9 +89,9 @@ export class APIState {
 	}
 
 	get coursesNotDeleted() {
-		const courses = [];
+		const courses: ICourseFromAPI[] = [];
 		for (const course of this.courses) {
-			if (!course.deletedAt) {
+			if (!course.deletedAt && !courses.some(existingCourse => `${existingCourse.subject}${existingCourse.crse}` === `${course.subject}${course.crse}`)) {
```

Comment:
this is quite an expensive check (O(n^2)), let's use a Set
also, could you provide an example of the bug? I haven't seen it before so curious what it looks like

Reply:
![Screenshot 2024-10-24 131449](https://github.com/user-attachments/assets/0b60b4d1-1a72-42d1-bc74-6c6352512793)
Here is the bug